cmd/kubeadm/app/phases/addons/dns/dns.go

```diff
@@ -57,7 +74,7 @@ func EnsureDNSAddon(cfg *kubeadmapi.MasterConfiguration, client clientset.Interf
 		ImageRepository: cfg.ImageRepository,
 		Arch:            runtime.GOARCH,
 		// Get the kube-dns version conditionally based on the k8s version
-		Version:        GetKubeDNSVersion(k8sVersion),
+		Version:        GetDNSVersion(k8sVersion, "kube-dns"),
```

Comment:
create a constant for `"kube-dns"` and `"coredns"` please in `constants.go` and use here.